### ISSUE TITLE
Permissions issues on linux

### ISSUE THREAD

**Issue by primeroz** [warrensbox/tgswitch#54]
Same issue as https://github.com/warrensbox/terraform-switcher/issues/33
```
$ tgswitch     
✔ 0.18.7
lstat /usr/local/bin/terragrunt: no such file or directory
symlink /home/user/.terragrunt.versions/terragrunt_0.18.7 /usr/local/bin/terragrunt: permission denied
2019/07/04 09:32:37 Unable to create symlink. You must have SUDO privileges symlink /home/user/.terragrunt.versions/terragrunt_0.18.7 /usr/local/bin/terragrunt: permission denied 
```

we need the same solution to use a configuration file to specify the destination bin link.

Thanks!

**Comment by Xat59** [comment]
I can confirm that behavior .

```
⇒  tgswitch                                                                                                                                                                  
✔ 0.19.21
symlink /home/xat/.terragrunt.versions/terragrunt_0.19.21 /usr/local/bin/terragrunt: file exists
2019/09/02 15:11:58 Unable to create symlink. You must have SUDO privileges symlink /home/xat/.terragrunt.versions/terragrunt_0.19.21 /usr/local/bin/terragrunt: file exists 
```

**Comment by warrensbox** [comment]
This feature was added with MR #65.
Upgrade tgswitch to get the latest version. 
```ruby
brew upgrade warrensbox/tap/tgswitch
```

## Please test the latest release and let me know if there are any bugs.
### Use custom installation location  (For non-admin - users with limited privilege on their computers)    
You can specify a custom binary path for your terragrunt installation

1. Create a custom binary path. Ex: `mkdir /Users/username/bin` (replace username with your username)
2. Add the path to your PATH. Ex: `export PATH=$PATH:/Users/username/bin` (add this to your bash profile or zsh profile)
3. Pass -b or --bin parameter with your custom path to install terragrunt. Ex: `tgswitch -b /Users/username/bin/terragrunt 0.14.1 `

**Comment by primeroz** [comment]
It does work but to be honest it would have been great to use the same way that "tfswitch" uses with the .toml file ... it is very likely that whoever use one tool use the other as well. 

In the case of tgswitch i will have to setup an alias to the `-b` option

**Comment by Xat59** [comment]
@primeroz I am surprised it fixed for you . Personnaly, I am not able to install it again with a limited privilege user :

```
$ curl -L https://raw.githubusercontent.com/warrensbox/tgswitch/release/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9129  100  9129    0     0  53075      0 --:--:-- --:--:-- --:--:-- 53075
warrensbox/tgswitch info checking GitHub for latest tag
warrensbox/tgswitch info found version: 0.4.326 for 0.4.326/linux/amd64
install: impossible de créer le fichier standard '/usr/local/bin/tgswitch': Permission non accordée
```

**Comment by primeroz** [comment]
@Xat59 what works for me is to have an alias and expand the PATH 

`tgswitch='$HOME/bin/tgswitch -b $HOME/bin/terragrunt'`

**Comment by Xat59** [comment]
@primeroz yeah it works after `tgswitch` reinstallation like I said here : https://github.com/warrensbox/tgswitch/issues/54#issuecomment-791280909

@warrensbox maybe the installation instructions must be reviewed for unprivileged users like I said before

**Comment by warrensbox** [comment]
I'll add the toml file option

**Comment by Deltaxzdev** [comment]
siempre que intento el ./install.sh el zsh me da permission denied que pueod hacer para arreglarlo
chmod -x install.sh             
                                                                                                                           
┌──(kali㉿kali)-[~/Desktop/DARKARMY]
└─$ ./install.sh 
zsh: permission denied: ./install.sh

**Comment by kvendingoldo** [comment]
btw. you can also use [tenv](https://github.com/tofuutils/tenv) that support Terraform as well as OpenTofu (and Terragrunt :) ) in one tool. It allow you to simplify version management and can do much more, than tgswitch.
